### PR TITLE
made the paddle move faster

### PR DESCRIPTION
**Pull request by rbral**

### CODE REVIEW

File: src/main/java/bwi/brickbreaker/Controller.java [hunk 1]
```diff
@@ -43,9 +44,38 @@ public void startGame() {
...
         view.repaint();
     }
 
+    private void checkIfGameOver() {
+        for (Brick brick : bricks) {
+            if (!brick.getBroken()) {
+                gameWon = false;
+                break;
+            }
+            if (gameWon) {
```
Comment (andrewoid) [inline src/main/java/bwi/brickbreaker/Controller.java]: Move this out of the for loop

File: src/main/java/bwi/brickbreaker/Controller.java [hunk 2]
```diff
@@ -66,27 +96,64 @@ public void checkPaddleCollision() {
...
+
+        // Determine the minimum distance to each side of the brick
+        double distanceToLeft = Math.abs(ballCenterX - brickLeft);
+        double distanceToRight = Math.abs(ballCenterX - brickRight);
+        double distanceToTop = Math.abs(ballCenterY - brickTop);
+        double distanceToBottom = Math.abs(ballCenterY - brickBottom);
+
+        // Check if the collision is vertical or horizontal by finding the smallest distance
+        if (distanceToTop < distanceToLeft && distanceToTop < distanceToRight && distanceToTop < distanceToBottom) {
+            return "horizontal"; // Top side collision
```
Comment (andrewoid) [inline src/main/java/bwi/brickbreaker/Controller.java:148]: Don't use Strings, use a constant (like an int or enum)

File: src/test/java/bwi/brickbreaker/GameTests.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        ArrayList<Brick> bricks = new ArrayList<>();
+        BoardComponent view = new BoardComponent(ball, paddle, bricks);
+        BrickBreakerModel model = new BrickBreakerModel(ball, bricks);
+        Controller controller = new Controller(ball, paddle, bricks, model, view);
+
+        //when
+        double startX = ball.getX();
+        double startY = ball.getY();
+        controller.startGame();
+        Thread.sleep(50);
```
Comment (andrewoid) [inline src/test/java/bwi/brickbreaker/GameTests.java:27]: Why are you sleeping?

**Comment by rbral** [comment]
Dear Professor Schwimmer, 
The code was edited according to your comments in the following PR: https://github.com/AImani2/brick_breaker/pull/12

**Pull request closed without merging**